core/Node.js

```diff
@@ -496,6 +496,41 @@ Node.prototype.getPosition = function getPosition () {
 };
 
 /**
+ * Determines the node's position relative to an ancestor node.
+ *
+ * @method getPositionFrom
+ *
+ * @param {Node} ancestor An ancestor `Node` of the current `Node`.
+ *
+ * @return {Float32Array} An array representing the position.
+ */
+Node.prototype.getPositionFrom = function getPositionFrom (ancestor) {
+    var position = Array.prototype.slice.call(this.getPosition());
+    var currentParent = this.getParent();
```

Comment:
Using `this.getParent()` instead of `this._parent`. You said not to access private properties, but, `this._parent` is part of this same class that this method is being defined on, so isn't it fine to use `this._parent`? Or should we change all other uses of `this._parent` to `this.getParent()`?

Oh, are you talking about only when accessing `._parent` on the `ancestor`? In that case I think it's fine. You can do this in Java, as long as the instance being accessed is the same type as the instance that's doing the access (hence the check for `instanceof Node`).

But using getParent() everywhere might be nicer in case the underlying logic changes.